add realtime statistic stdout and add cmake install

soyeonBak opened the pull request.

soyeonBak committed:
add realtime statistic stdout and add cmake install

soyeonBak committed:
Change realtimeFunc used as function parameter to typedef

Commit:
Apply clang-format

Commit:
A bit of tyding up

kwanglink commented on src/CMakeLists.txt:
```diff
@@ -65,3 +65,7 @@ endif()
   target_link_libraries(regexbench ${RE2_LIB})
 endif()
+
+file(GLOB RB_HDR_SRCS "*.h")
```
Isn't it better to limit the number of include files to be open public?

soyeonBak replied:
I'll limit it to what I need.

kwanglink commented on src/regexbench.h:
```diff
@@ -48,13 +49,23 @@ namespace regexbench {
-  size_t nmatched_pkts;
+  struct ResultInfo cur;
+  struct ResultInfo old;
+  bool stop;
```
How about declaring this as `std::atomic_bool` and use corresponding API's where it's being used.

soyeonBak replied:
I'll try to apply it.

soyeonBak committed:
1. limit the number of include files to be open public.
2. change bool to std::atomic_bool.

soyeonBak committed:
if quiet mode, disable realtime report.
getEngineType Function separation. When engine name is input, output engine number.

soyeonBak committed:
install include Rule.h Logger.h

soyeonBak committed:
fix library link error
- Integrate librb.a and libengines.a into librebench.

Commit:
Make it compile in linux
